autoload/slime/targets/neovim.vim

```diff
@@ -1,102 +1,226 @@
+  let jobid = getbufvar(buf_in, "&channel")
+  let job_pid = jobpid(jobid)
+
   if !exists("g:slime_last_channel")
-    let g:slime_last_channel = [{'jobid': &channel, 'pid': jobpid(&channel)}]
+    let g:slime_last_channel = [{'jobid': jobid, 'pid': job_pid, 'bufnr': buf_in}]
   else
-    call add(g:slime_last_channel, {'jobid': &channel, 'pid': jobpid(&channel)})
+    call add(g:slime_last_channel, {'jobid': jobid, 'pid': job_pid, 'bufnr': buf_in})
   endif
 endfunction
 
-function! slime#targets#neovim#SlimeClearChannel()
-  let current_buffer_jobid = get(b:,"terminal_job_id",-1)
-
+function! slime#targets#neovim#SlimeClearChannel(buf_in) abort
   if !exists("g:slime_last_channel")
-    if exists("b:slime_config")
-      unlet b:slime_config
-    endif
+    call s:clear_all_buffs()
     return
-  elseif len(g:slime_last_channel) == 1
+  elseif len(g:slime_last_channel) <= 1
+    call s:clear_all_buffs()
     unlet g:slime_last_channel
-    if exists("b:slime_config")
-      unlet b:slime_config
-    endif
   else
-    let bufinfo = s:get_filter_bufinfo()
+    let jobid_to_clear = filter(copy(g:slime_last_channel), {_, val -> val['bufnr'] == a:buf_in})[0]['jobid']
+    call s:clear_related_bufs(jobid_to_clear)
+    call filter(g:slime_last_channel, {_, val -> val['bufnr'] != a:buf_in})
+  endif
+endfunction
 
-    " tests if using a version of Neovim that
-    " doesn't automatically close buffers when closed
-    " or there is no autocommand that does that
-    if len(bufinfo) == len(g:slime_last_channel)
-      call filter(bufinfo, {_, val -> val != current_buffer_jobid})
+"evaluates whether ther is a terminal running; if there isn't then no config can be valid
+function! slime#targets#neovim#ValidEnv() abort
+  echohl WarningMsg
+  if (!exists("g:slime_last_channel") || (len(g:slime_last_channel)) < 1) || empty(g:slime_last_channel)
+    echo "Terminal not found."
+    echohl none
+    return 0
+  endif
+    echohl none
+  return 1
+endfunction
+
+" "checks that a configuration is valid
+" returns boolean of whether the supplied config is valid
+function! slime#targets#neovim#ValidConfig(config, silent) abort
+
+  echohl WarningMsg
+  if (!exists("g:slime_last_channel") || (len(g:slime_last_channel)) < 1) || empty(g:slime_last_channel)
+    if !a:silent
+      echo "Terminal not found."
     endif
+    echohl none
+    return 0
+  endif
 
-    call filter(g:slime_last_channel, {_, val -> index(bufinfo, str2nr(val["jobid"])) >= 0})
 
+  " Ensure the config is a dictionary and a previous channel exists
+  if type(a:config) != v:t_dict
+    if !a:silent
+      echo "Config type not valid."
+    endif
+    echohl none
+    return 0
+  endif
+
+  if empty(a:config)
+    if !a:silent
+      echo "Config is empty."
+    endif
+    echohl none
+    return 0
+  endif
+
+  " Ensure the correct keys exist within the configuration
+  if !(has_key(a:config, 'jobid'))
+    if !a:silent
+      echo "Configration object lacks 'jobid'."
```

Comment:
```suggestion
      echo "Configuration object lacks 'jobid'."
```

elsewhere you use "Config" in error message (consistence?)

Reply:
Fixed with commit, made consistent.